docs/spec/consensus/light-client.md

```diff
@@ -1,113 +1,313 @@
+the function *CheckSupport* in the process of
+bisection/recursion. *CheckSupport* implements the trusted period method and, for two adjacent headers (in term of heights), it checks uninterrupted sequence of proof.
+
+*Assumption*: In the following, we assume that *h2.Header.height > h1.Header.height*. We will quickly discuss the other case in the next section.
+
+We consider the following set-up:
+- the lite client communicates with one full node
+- the lite client locally stores all the signed headers it obtained (trusted or not). In the pseudo code below we write *Store(header)* for this.
+- If *Bisection* returns *false*, then the lite client has seen a  forged header.
+  * However, it does not know which header(s) is/are the problematic one(s).
+  * In this case, the lite client can submit (some of) the headers it has seen as evidence. As the lite client communicates with one full node only when executing Bisection, it is safe to assume that this full node is faulty.
+- the lite client must retry to retrieve correct headers from another full node
+  * it picks a new full node
+  * it restarts *Bisection*
+  * there might be optimizations; a lite client may not need to call *Commit(k)*, for a height *k* for which it already has a signed header it trusts.
+
+**Auxiliary Functions.** We will use the  function ```votingpower_in(V1,V2)``` to compute the voting power the validators in set V1 have according to their voting power in set V2;
+ ```votingpower_in(V,V)``` returns the total voting power in V.
+We further use the function ```signers(Commit)``` that returns the set of validators that signed the Commit.
+
+**CheckSupport.** The following function checks whether we can trust the header h2 based on header h1 following the trusting period method.
+
+```go
+  func CheckSupport(h1,h2,trustlevel,trustlevel_adj) bool {
+    if h1.Header.bfttime + tp < now { // Observation 1
+      return false // old header was once trusted but it is expired
+    }  
+    vp_all := votingpower_in(h1.Header.NextV,h1.Header.NextV)
+      // total sum of voting power in h1
+
+    if h2.Header.height == h1.Header.height + 1 {
+      // specific check for adjacent headers; everything must be
+      // properly signed.
+      // also check that h2.Header.V == h1.Header.NextV
+      // Plus the following check that 2/3 of the voting power
+      // in h1 signed h2
+      return (votingpower_in(signers(h2.Commit),h1.Header.NextV) >
+              max(2/3,trustlevel_adj) * vp_all)
+        // signing validators are more than two third in h1.
+        // headers in Cosmos are typically signed by all validators, so one
+        // may request that all signed (liveness might be endangered)
+    }
+
+    s := intersection(h1.Header.NextV,signers(h2.Commit))
+      // get validators in h1 that signed h2
+    vps := votingpower_in(s,h1.Header.NextV)
+      // sum of voting powers in h1 of
+      // validators that signed h2
+    return (vps > max(1/3,trustlevel) * vp_all)
```

Comment:
I assume we do more verification here before we return true if `(vps > max(1/3,trustlevel) * vp_all)` holds.
Should still check signatures and more than 2/3rds signed, like in the sequential case. This could be done in the beginning for both cases.

Reply:
Yes. When we receive a header we should do all checks to verify that the header is "well-formed". Is there a list of checks that we do in the sequential case that I could point to?

Reply:
https://github.com/tendermint/tendermint/blob/master/types/validator_set.go#L591-L633

There are other checks outside this when we receive the `FullCommit` but I think it would be enough if we add a comment that for both sequential and non-sequential cases we do the same basic verification.

Reply:
Thanks!